microWorker.js

```diff
@@ -194,6 +197,101 @@ function _setupFiles(bag, next) {
   );
 }
 
+function _setExecutorAsReqProc(bag, next) {
+  if (bag.jobStatusCode) return next();
+
+  var who = bag.who + '|' + _setExecutorAsReqProc.name;
+  logger.verbose(who, 'Inside');
+
+  bag.consoleAdapter.openCmd('Setting executor as reqProc');
+
+  var whoPath = util.format('%s/job.who', bag.buildStatusDir);
+  fs.writeFile(whoPath, 'reqProc\n',
+    function (err) {
+      if (err) {
+        var msg = util.format('%s, Failed to write file: %s ' +
+          'with err: %s', who, whoPath, err);
+        bag.consoleAdapter.publishMsg(msg);
+        bag.consoleAdapter.closeCmd(false);
+        bag.consoleAdapter.closeGrp(false);
+        bag.jobStatusCode = __getStatusCodeByName('error', bag.isCI);
+        return next();
+      }
+
+      bag.consoleAdapter.publishMsg(
+        util.format('Updated %s', whoPath)
+      );
+      bag.consoleAdapter.closeCmd(true);
+      return next();
+    }
+  );
+}
+
+function _generateSteps(bag, next) {
+  if (bag.jobStatusCode) return next();
+
+  var who = bag.who + '|' + _generateSteps.name;
+  logger.verbose(who, 'Inside');
+
+  bag.consoleAdapter.openCmd('Generating job steps');
+
+  // TODO: job steps are being read from example file temporarily
+  // This section will be replaced by actual generation of job steps in future
+
+  var exampleSteps = util.format('%s/%s', __dirname,
```

Comment:
I think, i will use the `_common/example/steps.json` directly in the first argument.